test/src/problems/FixedPointProblem.C

```diff
@@ -18,43 +18,65 @@ FixedPointProblem::validParams()
 {
   InputParameters params = FEProblem::validParams();
   params.addParam<TagName>("fp_tag_name", "fp_previous", "Tag name for the fixed point iteration");
+  params.addParam<bool>(
+      "tagged_vector_for_partial_residual",
+      true,
+      "Toggle between parital residual and previous solution for the tagged vector");
   return params;
 }
 
 FixedPointProblem::FixedPointProblem(const InputParameters & params)
   : FEProblem(params),
+    _tagged_vector_for_partial_residual(getParam<bool>("tagged_vector_for_partial_residual")),
     _tag_previous(getParam<TagName>("fp_tag_name")),
-    _tag_id(addVectorTag(_tag_previous)),
-    _residual_previous(_nl->addVector(_tag_id, false, GHOSTED))
+    _tag_id(addVectorTag(_tag_previous,
+                         _tagged_vector_for_partial_residual ? Moose::VECTOR_TAG_RESIDUAL
+                                                             : Moose::VECTOR_TAG_SOLUTION)),
+    _tagged_vector(_nl->addVector(_tag_id, false, GHOSTED))
```

Comment:
IIRC, you do not like ghosted vector:-) Could we use parallel vector to save memory ?

Reply:
I did not try in MOOSE but did try in Griffin and failed. I will try and let you know.

Reply:
Just tried, these two new tests using tagged solution vector:
```
executioners/fixed_point.tagged_solution_vector/var ........................................... FAILED (CRASH)
executioners/fixed_point.tagged_solution_vector/material ...................................... FAILED (CRASH)
```
crashed with four processors. I guess for nodal variables, we have to make it ghosted. For DG, DG kernels typically require ghosted solutions as well. If the tagged vector is for residual, it can be less restrictive unless the vector is used as solutions. Conclusion is that we have to make this ghosted.